core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -79,7 +85,12 @@ public static ManifestReader read(InputFile file) {
    * @return a manifest reader
    */
   public static ManifestReader read(InputFile file, Map<Integer, PartitionSpec> specsById) {
-    return new ManifestReader(file, specsById);
+    return new ManifestReader(file, specsById, Predicates.alwaysTrue());
+  }
+
+  public static ManifestReader read(
+      InputFile file, Map<Integer, PartitionSpec> specsById, Predicate<ManifestEntry> manifestEntryPredicate) {
```

Comment:
I don't think this is the right place to add a predicate. This class is used to either get the entire list of entries by calling `entries` or to get the entire list of files by using it as an `Iterator`.

All filtering should be done using the refinement methods, like `filterPartitions` or `filterRows`. Some classes, like [`ManifestGroup`](https://github.com/apache/incubator-iceberg/blob/master/core/src/main/java/org/apache/iceberg/ManifestGroup.java#L178-L202) use the reader and filters to get entries and then add additional predicates.

I think we should take that approach instead of changing how to use `ManifestReader`.